infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/InSnapshotSyncState.java

```diff
@@ -152,6 +153,9 @@ public void onEntry(LogReplicationState from) {
         try {
             // If the transition is to itself, the snapshot sync is continuing, no need to reset the sender
             if (from != this) {
+                if (from.getType().equals(LogReplicationStateType.INITIALIZED)) {
```

Comment:
having this check will be a problem - when in log entry sync and sync_cancel is received, we enter this state.  This validation on the 'from' state will prevent the sync type from getting updated and markSnapshotSyncInfoOngoing() will have the wrong sync type.